### PR TITLE
Check to see if build output has been committed

### PR DESCRIPTION
**Pull request by LucianBuzzo**
Connects to #46

The doc generation script has been refactored to be more platform agnostic,
allowing it to be run on the Travis servers

change-type: patch

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,10 +15,10 @@
     "semver"
   ],
   "scripts": {
-    "test": "npm run build && npm run lint && mocha -r ts-node/register test/index.spec.ts",
+    "test": "npm run build && npm run lint && ./automation/test.js",
     "build": "tsc",
     "lint": "tslint -t stylish -c tslint.json 'src/**/*.ts' 'test/**/*.ts'",
-    "docs": "npm run build && node generateDocs"
+    "docs": "npm run build && ./automation/docs.js"
```
Comment (pimterry) [inline package.json]: This is preexisting, but a bit odd, and doesn't line up with how our other projects work. Let's fix it: if you stop `docs` running `build`, and make `build` run `docs` instead, then `npm run build` will build _everything_, as people will expect, and the `catch-uncommitted-output.sh` script can just be identical to every other project's.

File: package.json [hunk 2]
```diff
@@ -15,10 +15,10 @@
     "semver"
   ],
   "scripts": {
-    "test": "npm run build && npm run lint && mocha -r ts-node/register test/index.spec.ts",
+    "test": "npm run build && npm run lint && ./automation/test.js",
```
Comment (pimterry) [inline package.json]: What does having the mocha script in a separate file add?

Unless there's a nice benefit, I'd rather keep it simple and do it inline here (might be nice to change the path to `test/*.spec.ts` though)

File: package.json [hunk 3]
```diff
@@ -32,10 +32,11 @@
   },
   "devDependencies": {
     "chai": "^4.1.0",
+    "intercept-stdout": "^0.1.2",
```
Comment (pimterry) [inline package.json]: Why is this needed?

File: package.json (hunk 2 above)
Comment (LucianBuzzo) [reply package.json]: The doc generation relies on knowing some information about how the mocha test runs, rather than parsing the test script with a regex (like we did before) I've made mocha run programmatically and use a shareable configuration file. It might seem silly to do this since it's only sharing a test target at the moment, however, I anticipate there being a point in the future where we want to add more complex options to mocha and it breaks the docs generation.

File: package.json (hunk 1 above)
Comment (LucianBuzzo) [reply package.json]: Yep this sounds good, thanks for detangling my spaghetti! :smile:

File: package.json (hunk 3 above)
Comment (LucianBuzzo) [reply package.json]: This was a failed experiment and needs to be removed.

**Comment by LucianBuzzo** [comment]
We should switch to using the new `catch-uncommitted module` https://github.com/resin-io-modules/catch-uncommitted

File: package.json [hunk 4]
```diff
@@ -15,10 +15,11 @@
...
   ],
   "scripts": {
-    "test": "npm run build && npm run lint && mocha -r ts-node/register test/index.spec.ts",
-    "build": "tsc",
+    "test": "npm run build && npm run lint && ./automation/test.js",
+    "build": "tsc && npm run docs",
     "lint": "tslint -t stylish -c tslint.json 'src/**/*.ts' 'test/**/*.ts'",
-    "docs": "npm run build && node generateDocs"
+    "docs": "./automation/docs.js",
+    "ci": "catch-uncommitted && npm test"
```
Comment (pimterry) [inline package.json]: `catch-uncommitted` needs to come _after_ the `npm test` - it now doesn't run the build, it just checks the state (so you can run the build however you like), so it needs to happen after all the build & docs & testing etc has happened.

Comment (LucianBuzzo) [reply package.json]: ah I missed that!

Comment (pimterry) [reply package.json]: To be fair, I did just silently change it from the previous script behaviour without mentioning anything. This one's definitely my fault :smiley:. Sorry!

Comment (LucianBuzzo) [reply package.json]: I moved things around and it should be good now
